packages/app/components/SelectNetwork.tsx

```diff
@@ -1,29 +1,85 @@
 "use client";
 
-import { Button, Icon } from "@/ui";
-import { useAccount, useNetwork, useSwitchNetwork } from "wagmi";
+import { Fragment, useEffect, useState } from "react";
+
+import type { Chain } from "viem/chains";
 import { ChainIcon } from "connectkit";
 import { Listbox, Transition } from "@headlessui/react";
-import { Fragment } from "react";
-import { parseAsInteger, useQueryState } from "next-usequerystate";
+import { useAccount, useNetwork, useSwitchNetwork } from "wagmi";
+
+import { Button, Icon } from "@/ui";
+import { config } from "@/providers";
+import { useStackboxFormContext } from "@/contexts";
+
+interface WagmiChain extends Chain {
+  unsupported?: boolean;
+}
 
 export const SelectNetwork = () => {
-  const [, setChainId] = useQueryState("chainId", parseAsInteger);
+  const { stackboxFormState } = useStackboxFormContext();
+  const [, setChainId] = stackboxFormState.chainIdState;
   const { switchNetwork } = useSwitchNetwork({
     onSuccess(data) {
       setChainId(data.id);
     },
   });
-  const { chain, chains } = useNetwork();
+  const { chain } = useNetwork();
   const { isConnected } = useAccount();
+  const [selectedChain, setSelectedChain] = useState<WagmiChain>();
+  const [allowedChains, setAllowedChains] = useState<WagmiChain[] | undefined>(
+    []
+  );
+
+  const handleOfflineNetworkSwitch = (newChainId: number) => {
```

Comment:
I don't think offline represents not connected. I would rename it.

Eg. `handleDisconnectedNetworkSwitch` or `handleNetworkSwitchWhenNotConnected`

```suggestion
  const handleDisconnectedNetworkSwitch = (newChainId: number) => {
```